clouddriver-cf/src/main/groovy/com/netflix/spinnaker/clouddriver/cf/config/CloudFoundryConfig.groovy

```diff
@@ -37,9 +44,53 @@ import org.springframework.scheduling.annotation.EnableScheduling
 class CloudFoundryConfig {
 
 	@Bean
-	@ConfigurationProperties("cf")
-	CloudFoundryConfigurationProperties cloudFoundryConfigurationProperties() {
-		new CloudFoundryConfigurationProperties()
+	CloudFoundryConfigurationProperties cfConfigurationProperties() {
+		new CloudFoundryConfigurationProperties();
+	}
+
+	@Bean
+	CloudFoundryCredentialsInitializer cloudFoundryCredentialsInitializer() {
+		new CloudFoundryCredentialsInitializer();
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(CloudFoundryClientFactory)
+	CloudFoundryClientFactory cloudFoundryClientFactory() {
+		new DefaultCloudFoundryClientFactory()
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(CloudFoundryDeployHandler)
+	DeployHandler deployHandler(CloudFoundryClientFactory clientFactory) {
```

Comment:
this bean should probably get a different name (`cloudFoundryDeployHandler`?) - it seems ripe for one of those hard to diagnose spring context init problems when someone else drops in a bean of the same name